src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,162 @@
+package duke;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import duke.exception.DukeException;
+import duke.helper.Parser;
+import duke.helper.Storage;
+import duke.helper.TaskList;
+import duke.helper.Ui;
+import java.io.IOException;
+
+public class Duke extends Application {
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    /**
+     * The main function for Duke, which runs the program.
+     *
+     * @param args commandline args.
+     * @throws IOException
```

Comment:
A suggestion would be to explain in what circumstance an IOException will be thrown, as shown in this example [here](https://nus-cs2103-ay2223s2.github.io/website/book/documentation/tools/javaDoc/how/). This would make the documentation more complete.